tests/common/plugins/sanity_check/__init__.py

```diff
@@ -416,17 +374,27 @@ def recover_on_sanity_check_failure(duthosts, failed_results, fanouthosts, local
     new_failed_results = [result for result in new_check_results if result['failed']]
     if new_failed_results:
         request.config.cache.set(cache_key, True)
-        request.config.cache.set(recovery_cache_key, False)
+        request.config.cache.set(custom_msg_prefix + cache_key, True)
+        request.config.cache.set(custom_msg_prefix + recovery_cache_key, False)
         pt_assert(False,
                   f"!!!!!!!!!!!!!!!! {sanity_check_stage} sanity check after recovery failed: !!!!!!!!!!!!!!!!\n"
                   f"{json.dumps(new_failed_results, indent=4, default=fallback_serializer)}")
     # Record recovery success
     request.config.cache.set(recovery_cache_key, True)
+    request.config.cache.set(custom_msg_prefix + recovery_cache_key, True)
 
 
-# make sure teardown of log_custom_msg happens after sanity_check
 @pytest.fixture(scope="module", autouse=True)
-def sanity_check(request, parallel_run_context, log_custom_msg):
+def sanity_check(request, parallel_run_context):
+    # reset all the sonic_custom_msg keys from cache
+    # reset here because this fixture will always be very first fixture to be called
+    cache_dir = request.config.cache._cachedir
```

Comment:
Let's put this logic in the hook of pytest starting, 
There should be pytest_sessionstart just like pytest_sessionfinish, please confirm and try